File: benches/mutations/variants/python/seed3.ctl-tree-recursion.py
def reduce_pair(a, b, log):
    if b == 0:
        return a
    log.append("step: " + str(a) + " " + str(b))
    left = reduce_pair(b, a % b, log)
    right = reduce_pair(b % 3, a, log)
    return max(left, right)

File: benches/mutations/variants/rust/seed3.ctl-tree-recursion.rs
fn reduce_pair(a: u64, b: u64, log: &mut Vec<String>) -> u64 {
    if b == 0 {
        return a;
    }
    log.push(format!("step: {a} {b}"));
    let left = reduce_pair(b, a % b, log);
    let right = reduce_pair(b % 3, a, log);
    left.max(right)
}
